File: be_hotel/src/main/java/be_hotel/serive/room/RoomService.java
package be_hotel.serive.room;

import be_hotel.Modal.entity.Room;
import be_hotel.Modal.request.RoomRequest;
import org.springframework.data.domain.Page;

public interface RoomService {
    public Room createRoom(RoomRequest roomRequest);
    Page<Room> getAllRoom(int page, int size);
    public Room updateRoom(RoomRequest roomRequest, Integer id);
    public String deleteRoom(Integer id);
}

File: src/services/RoomService.js
import api from "./api";

const RoomService = {
    getRoom(page, size){
        return api.get(`getAllRoom/${page}/${size}`)
    },
    createRoom(body){
        return api.post("createRoom", body)
    },
    updateRoom(id, body){
        return api.put("updateRoom/"+id, body)
    },
    deleteRoom(id){
        return api.delete("deleteRoom/"+id)
    }
}
export default RoomService;